### PR TITLE
Update scalatest to 3.0.7

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: akka-actor-testkit-typed/src/test/java/akka/actor/testkit/typed/javadsl/ActorTestKitTest.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import akka.actor.typed.javadsl.Behaviors;
 import org.junit.ClassRule;
 import org.junit.Test;
-import org.scalatestplus.junit.JUnitSuite;
+import org.scalatest.junit.JUnitSuite;
```
Comment (jlprat) [inline akka-actor-testkit-typed/src/test/java/akka/actor/testkit/typed/javadsl/ActorTestKitTest.java]: This class is deprecated and we should use the one under `scalatestpulus`. However, the latter doesn't seem to be "importable". For reference: https://github.com/akka/akka-http/pull/2475 and  https://github.com/scalatest/scalatest/issues/1553

**Review by jlprat — APPROVED** [review]
LGTM, I added the notes about the deprecation and the links to the Akka HTTP discussion and the ScalaTest issue

File: project/Dependencies.scala [hunk 2]
```diff
@@ -29,7 +29,7 @@ object Dependencies {
         case Some((2, n)) if n >= 12 => "1.14.0" // does not work for 2.11
         case _                       => "1.13.2"
       }),
-    scalaTestVersion := "3.0.6-SNAP6",
+    scalaTestVersion := "3.0.7",
```
Comment (helena) [inline project/Dependencies.scala]: 👍 Finally off SNAP.

File: akka-actor-testkit-typed/src/test/java/akka/actor/testkit/typed/javadsl/ActorTestKitTest.java (hunk 1 above)
Comment (helena) [reply akka-actor-testkit-typed/src/test/java/akka/actor/testkit/typed/javadsl/ActorTestKitTest.java]: In the `3.0.7` tag they are still using `org.scalatestplus.junit.JUnitSuite`.

**Review by patriknw — APPROVED** [review]
LGTM, if we have verified that the java tests are actually running and only once
(we have many problems with that over the years)

File: akka-actor-testkit-typed/src/test/java/akka/actor/testkit/typed/javadsl/ActorTestKitTest.java (hunk 1 above)
Comment (raboof) [reply akka-actor-testkit-typed/src/test/java/akka/actor/testkit/typed/javadsl/ActorTestKitTest.java]: They're still referring to it from the 'deprecated' annotation, but they don't seem to be actually shipping it...

**Review by chbatey — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
checked jenkins results/logs, all good

**Pull request merged**